enhancements/storage/csi-driver-install.md

```diff
@@ -0,0 +1,196 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/**
+
+## Summary
+
+We want certain CSI drivers such as AWS, GCE, Cinder, Azure and vSphere to be installable on Openshift, so as
+they can be used along-side in-tree drivers and when upstream enables migration flag for these volume types, their
+replacement CSI drivers can take over and none of the storage features get affected.
+
+## Motivation
+
+Upstream Kubernetes is moving towards removing code of in-tree drivers and replacing them with their CSI counterpart. Our
+current expectation is that - all in-tree drivers that depend on cloudprovider should be removed from core Kubernetes by 1.21.
+This may not happen all at once and we expect migration for certain drivers to happen sooner.
+
+This does mean that - Openshift should be prepared to handle such migration. We have to iron out any bugs in driver themselves and
+their interfacing with Openshift. We need a way for users to use the CSI drivers and optionally enable migration from in-tree driver
+to CSI driver. To support upstream design - we will also need a way for users to disable the migration and keep using in-tree driver, until
+in-tree code is finally removed.
+
+## Terminology
+
+CSI migration - seamless migration of in-tree volume plugins (AWS EBS, GCE PD, OpenStack Cinder, vSphere, Azure Disk, Azure File) to their CSI counterparts. In a cluster with CSI migration enabled for a particular plugin, Kubernetes translates all in-tree volume plugin calls into CSI under the hood, without users changing their PVs / StorageClasses. This enables removal of the cloud specific code and whole cloud providers out of kubernetes/kubernetes.
+
+## Goals
+
+We would like to approach various goals of this KEP in different phases, because we do not expect to get everything right or done in phase-1.
+
+### Phase-1 Goals
+
+* Provide a way for AWS and GCE CSI driver installation . Users could either optionally install it or CSI driver could be installed
+along with in-tree driver and users could use both.
+* Install CSI provided storageclass along with in-tree StorageClass.
+
+### Phase-2 Goals
+
+* Support installation of GCE, Cinder, vSphere, AzureDisk CSI drivers.
+* Provide a way for users to enable in-tree to CSI migration. It basically replaces whole in-tree volume plugin with a less tested CSI implementation. While it's not a strict requirement, we'd like to have the migration optional for at least one release.
+* Provide a way for users to disable in-tree to CSI migration.
+
+### Phase-3 Goals
+
+* Enable CSI drivers as default drivers and CSI provided storageclass as default storageclass.
+* Optionally allow users to configure CSI driver install.
+
+## Non-Goals
+
+* This KEP does not attempt to design installation of third-party or partner CSI drivers.
```

Comment:
Supposing we  (redhat) developed a CSI driver for mounting "global" secrets and configmaps or a CSI driver for mounting an image filesystem, would we use this installation method or a different one?  I'm not claiming this choice isn't right, I'm asking about whether other units of "we" are partners

Reply:
For OCP specific CSI drivers that are always installed, I would suggest an operator managed by CVO and avoid OLM.